### PR TITLE
Build budget categories by ID to handle top-level categories

### PR DESCRIPTION
**Pull request by tylerneu**
Here is how I handled my own budget utilizing top-level categories.

### CODE REVIEW

File: mintapi/api.py [hunk 1]
```diff
@@ -151,7 +151,10 @@ def get_accounts(self, get_detail=False):  # {{{
 
     def get_transactions(self):
         if not pd:
-            raise ImportError('transactions data requires pandas; please pip install pandas')
+            raise ImportError(
+                'transactions data requires pandas; '
+                'please pip install pandas'
+            )
```
Comment (mrooney) [inline mintapi/api.py]: It would be great to avoid unrelated changes. Is this not PEP 8 already?

File: setup.py [hunk 2]
```diff
@@ -6,7 +6,9 @@
     from pypandoc import convert
     read_md = lambda f: convert(f, 'rst')
 except ImportError:
-    print("warning: pypandoc module not found, could not convert Markdown to RST")
+    print(
+        "warning: pypandoc module not found, could not convert Markdown to RST"
+    )
```
Comment (mrooney) [inline setup.py]: Similarly, let's avoid this unless there is a compelling reason.

File: mintapi/api.py [hunk 3]
```diff
@@ -263,8 +266,6 @@ def get_categories(self):  # {{{
         return categories
 
     def get_budgets(self):  # {{{
-        # Get categories
-        categories = self.get_categories()
```
Comment (mrooney) [inline mintapi/api.py]: I think it would be best to keep this here, so we can pass it to get_category_from_id, otherwise it is calling self.get_categories() quite a few times for no particular reason. Even if it happens to be a no-op after the first call, that's still an implementation detail that we shouldn't assume :)

File: mintapi/api.py [hunk 4]
```diff
@@ -293,10 +294,27 @@ def get_budgets(self):  # {{{
         # Fill in the return structure
         for direction in budgets.keys():
             for budget in budgets[direction]:
-                budget['cat'] = categories[budget['cat']]
+                budget['cat'] = self.get_category_from_id(budget['cat'])
 
         return budgets
 
+    def get_category_from_id(self, cid):
```
Comment (mrooney) [inline mintapi/api.py]: Take categories as an argument here to avoid self.get_categories().

**Comment by mrooney** [comment]
Thanks so much for your time Tyler! If you could just address the few comments, I think this would be ready to go :)

File: mintapi/api.py (hunk 1 above)
Comment (tylerneu) [inline mintapi/api.py]: I agree but [it wasn't PEP 8.](https://travis-ci.org/mrooney/mintapi/jobs/84168110)

File: setup.py (hunk 2 above)
Comment (tylerneu) [inline setup.py]: Same as above. I only made this change to satisfy my own execution of `flake8` but in retrospect Travis CI never complained. I'll back this out.

File: mintapi/api.py (hunk 3 above)
Comment (tylerneu) [inline mintapi/api.py]: I understand and agree with your suggestion.

**Comment by tylerneu** [comment]
I have finished my changes based on your comments. Let me know if there is anything else I can do.

File: mintapi/api.py (hunk 1 above)
Comment (mrooney) [inline mintapi/api.py]: Apologies, thanks!
